app/survey_translator.py

```diff
@@ -0,0 +1,245 @@
+
+    def _create_translation_object(self, translation_file):
+        _, translation_ext = os.path.splitext(translation_file)
+
+        if translation_ext not in ['.po', '.csv']:
+            print('** ERROR ** We only support po or csv files for translation')
+        elif translation_ext == '.po':
+            self.create_translation_object_from_pot(translation_file)
+        elif translation_ext == '.csv':
+            self.create_translation_object_from_csv(translation_file)
+
+    def create_translation_object_from_pot(self, pot_file):
+        msgctxt = None
+        value = ''
+        with open(pot_file) as f:
+            for line in f:
+                if line.startswith('msgctxt'):
+                    msgctxt = line.split('"')[1]
+                elif line.startswith('msgid'):
+                    value = line.split('"')[1]
+                elif line.startswith('msgstr'):
+                    object_key = self.get_object_key(value, msgctxt)
+                    if value != "":
+                        translation = line.split('"')[1]
+                        self.translation_object[object_key] = dumb_to_smart_quotes(translation)
+                    msgctxt = None
+
+    def create_translation_object_from_csv(self, pot_file):
+        for line in [line.rstrip('\n') for line in open(pot_file)]:
+            comma_separation = line.split(",")
+            msgctxt = comma_separation[0] if comma_separation[0] != 'Not Required' else None
+            english_text = comma_separation[1]
+            object_key = self.get_object_key(msgctxt, english_text)
+            translation = comma_separation[2]
+            self.translation_object[object_key] = dumb_to_smart_quotes(translation)
+
+    def get_object_key(self, value, msgctxt):
+        return (TemplateExtractor.trim_chars_from_object_key(value),
+                TemplateExtractor.trim_chars_from_object_key(msgctxt))
+
+    def validate_translation_object(self, translation_object):
+        """
+        Identify missing translation values within the SurveyTranslator
+        for a given template
+
+        :param translation_object:
+        :return:
+        """
+        for object_key in translation_object.keys():
+            if self.translation_object.get(object_key) is None:
+                missing_object = translation_object[object_key]
+                error_message = 'Missing translation for '
+                id = missing_object['context'][0]
+                for context in missing_object['context']:
+                    if 'id: ' in context:
+                        id = 'id:' + context.split(':')[1]
+                error_message += id + ' ~ "' + missing_object['value'] + '"'
+                self.missing_translations += [error_message]
+                print(error_message)
+
+    def translate_schema(self, data):
+        # *** TRANSLATE LEGAL_BASIS ***
+        if data.get('legal_basis') != 'Voluntary':
```

Comment:
Ran it directly in PyCharm. 
Getting `KeyError` on `legal_basis` so this could be something like this?
```py
	if data.get('legal_basis') and data.get('legal_basis') != 'Voluntary':
```